// react-facebook/lib/Page.js
import React from 'react';
import PropTypes from 'prop-types';
import Parser from './Parser';
import getCurrentHref from './utils/getCurrentHref';

export default function Page(props) {
  var className = props.className,
      style = props.style,
      _props$href = props.href,
      href = _props$href === undefined ? getCurrentHref() : _props$href,
      tabs = props.tabs,
      hideCover = props.hideCover,
      width = props.width,
      height = props.height,
      showFacepile = props.showFacepile,
      hideCTA = props.hideCTA,
      smallHeader = props.smallHeader,
      adaptContainerWidth = props.adaptContainerWidth,
      children = props.children,
      onParse = props.onParse;


  return React.createElement(
    Parser,
    { className: className, onParse: onParse },
    React.createElement(
      'div',
      {
        className: 'fb-page',
        style: style,
        'data-tabs': tabs,
        'data-hide-cover': hideCover,
        'data-show-facepile': showFacepile,
        'data-hide-cta': hideCTA,
        'data-href': href,
        'data-small-header': smallHeader,
        'data-adapt-container-width': adaptContainerWidth,
        'data-height': height,
        'data-width': width
      },
      children
    )
  );
}

Page.propTypes = {
  className: PropTypes.string,
  href: PropTypes.string.isRequired,
  tabs: PropTypes.string,
  hideCover: PropTypes.bool,
  height: PropTypes.oneOfType([PropTypes.number.isRequired, PropTypes.string.isRequired]),
  width: PropTypes.oneOfType([PropTypes.number.isRequired, PropTypes.string.isRequired]),
  showFacepile: PropTypes.bool,
  hideCTA: PropTypes.bool,
  smallHeader: PropTypes.bool,
  adaptContainerWidth: PropTypes.bool,
  children: PropTypes.node,
  onParse: PropTypes.func
};

Page.defaultProps = {
  width: 340,
  height: 500,
  tabs: 'timeline',
  hideCover: false,
  showFacepile: true,
  hideCTA: false,
  smallHeader: false,
  adaptContainerWidth: true,
  children: undefined,
  className: undefined,
  onParse: undefined
};
